source/MultiGrid/MultiGrid.js

```diff
@@ -549,22 +503,57 @@ export default class MultiGrid extends React.PureComponent {
     }
 
     if (
-      firstRender ||
+      resetAll ||
       leftSizeChange ||
-      styleTopRightGrid !== prevProps.styleTopRightGrid
+      styleTopRightGrid !== this._lastRenderedStyleTopRightGrid
     ) {
       this._topRightGridStyle = {
-        left: this._getLeftGridWidth(props),
+        left: this._getLeftGridWidth(this.props),
         overflowX: enableFixedRowScroll ? 'auto' : 'hidden',
         overflowY: 'hidden',
         position: 'absolute',
         top: 0,
         ...styleTopRightGrid,
       };
     }
+
+    this._lastRenderedColumnWidth = columnWidth;
+    this._lastRenderedFixedColumnCount = fixedColumnCount;
+    this._lastRenderedFixedRowCount = fixedRowCount;
+    this._lastRenderedHeight = height;
+    this._lastRenderedRowHeight = rowHeight;
+    this._lastRenderedStyle = style;
+    this._lastRenderedStyleBottomLeftGrid = styleBottomLeftGrid;
+    this._lastRenderedStyleBottomRightGrid = styleBottomRightGrid;
+    this._lastRenderedStyleTopLeftGrid = styleTopLeftGrid;
+    this._lastRenderedStyleTopRightGrid = styleTopRightGrid;
+    this._lastRenderedWidth = width;
   }
 
-  _onScroll(scrollInfo) {
+  _prepareForRender() {
```

Comment:
FWIW I'm a fan of this pattern & think it's much more readable than the large render functions that something like `CollectionView` has.